### DISCUSSION TITLE
file convert command as plugin

### DISCUSSION THREAD

**Discussion by woncheol-kim** [jarun/nnn#1554] [Q&A] [answered]
Hi, I've tried following command as plugin:

t:-!|ebook-convert $nnn $nnn.mobi

And I've got an error message like this:

Cannot read from (Current_Directory)/$nnn

ebook-convert is a CLI tool packaged with Calibre and what I actually want to do is like this:

ebook-convert sourcefile.txt sourcefile.mobi

What can I do to make it work properly?

**Comment by 0xACE** [comment]
Not sitting infront of a computer, but I dont remember if `nnn` gives arguments to the plugins.

you need to write a script to wrap the arguments to your liking

**Comment by jarun** [comment, marked as answer]
With the `|` (indicator to page non-interactive command output) removed, I could use the following:

    c:!cp "$nnn" "$nnn".cp*

I have updated the documentation accordingly.

Multiple arguments can be passed and variables can be used (including double-quoted ones) for all cases except running GUI app as plugin and paging the output of non-interactive commands.

**Comment by woncheol-kim** [comment]
Thank you. What's the dash before ! in many examples?

**Reply by jarun** [reply]
Please mark as answered of this solves your problem.

Plugin documentation: https://github.com/jarun/nnn/tree/master/plugins#nnn-plugins

**Reply by woncheol-kim** [reply]
Sorry, but I don't know how I can mark this solved.

**Reply by jarun** [reply]
There is a "Mark and answer" button below the each reply.

**Reply by woncheol-kim** [reply]
Thanks. I did it. :)
